### PR TITLE
Remove agent stop/disable/enable/start from RPM + 4.x -> 4.x Upgrades

### LINKED ISSUE #512: Upgrade from 4.x
We currently only test upgrading from the last 3.1.x release.

We should also test upgrading from the most recent 4.x release.

### PR DESCRIPTION
**Pull request by brianjmurrell**
Fixes #512

Have our upgrade test work from the most recent 4.x release.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

---
This change is [Reviewable](https://reviewable.io/reviews/intel-hpdd/intel-manager-for-lustre/513)

### CODE REVIEW

File: chroma-agent/chroma-agent.spec [hunk 1]
```diff
@@ -139,12 +133,17 @@ setenforce 0
...
-    systemctl start %{name}
-    # upgrade; convert any older agent config
+    # upgrade; convert any older agent config and then restart
     chroma-agent convert_agent_config
+    systemctl enable %{name}
+    # it's tempting to restart the agent here, but this upgrade
+    # could have been initiated from the running chroma-agent and
+    # so restarting here would pull the rug out from under that
+    # upgrade procedure
+    #systemctl restart %{name}
```
Comment (jgrund) [inline chroma-agent/chroma-agent.spec]: Do we need the commented out restart statement as well?

File: chroma-manager/tests/framework/utils/selective_auto_pass.sh [hunk 2]
```diff
@@ -64,10 +64,12 @@ check_for_autopass() {
     fi
 
     # RHEL 7.5 won't upgrade CentOS 7.3
-    if [[ ($JOB_NAME == upgrade-tests || $JOB_NAME == upgrade-tests/*) &&
-        $TEST_DISTRO_NAME != rhel ]]; then
-        fake_test_pass "upgrade-tests_skipped_on_centos7.3" "$WORKSPACE/test_reports/" "${BUILD_NUMBER}"
-        exit 0
-    fi
+    # still true?
+    #if [[ ($JOB_NAME == upgrade-tests || $JOB_NAME == upgrade-tests/*) &&
```
Comment (jgrund) [inline chroma-manager/tests/framework/utils/selective_auto_pass.sh]: Do we need this code still?

File: chroma-agent/chroma-agent.spec (hunk 1 above)
Comment (brianjmurrell) [reply chroma-agent/chroma-agent.spec]: Are you asking if we need it still commented out or if we need it uncommented?

Comment (jgrund) [reply chroma-agent/chroma-agent.spec]: Is it worth keeping the ```#systemctl restart %{name}``` there v.s. just removing it.

Comment (brianjmurrell) [reply chroma-agent/chroma-agent.spec]: I left it there as an example of what we don't want to do

File: chroma-manager/tests/framework/integration/installation_and_upgrade/run_tests [hunk 3]
```diff
@@ -362,14 +390,15 @@ TESTS="$TESTS_DIR/test_data_consistancy.py \
...
+export UPGRADE_FROM_3=$UPGRADE_FROM_3
 ./tests/integration/run_tests -f -c /root/cluster_cfg.json -x ~/test_report_post_upgrade.xml $TESTS"
 
 # now provide an information inventory of the difference in the RPM
 # catalog after the upgrade
 
 pdsh -l root -R ssh -S -w "$(spacelist_to_commalist "$ALL_NODES")" "exec 2>&1; set -xe
 if [ -f /tmp/rpms_before_upgrade ]; then
-    if ! diff -u /tmp/rpms_before_upgrade <(rpm -qa | sort); then
+    if ! diff -W 120 -y /tmp/rpms_before_upgrade <(rpm -qa | sort); then
```
Comment (tanabarr) [inline chroma-manager/tests/framework/integration/installation_and_upgrade/run_tests]: might be good to have some indication of what / why "120" is chosen here

**Pull request merged**